internal/storage/bucket_handle.go

```diff
@@ -311,3 +311,47 @@ func (b *bucketHandle) UpdateObject(ctx context.Context, req *gcs.UpdateObjectRe
 
 	return
 }
+
+func (b *bucketHandle) ComposeObjects(ctx context.Context, req *gcs.ComposeObjectsRequest) (o *gcs.Object, err error) {
+	dstObj := b.bucket.Object(req.DstName)
+
+	if req.DstGenerationPrecondition != nil {
+		dstObj = dstObj.If(storage.Conditions{GenerationMatch: *req.DstGenerationPrecondition})
+	}
+	if req.DstMetaGenerationPrecondition != nil {
+		dstObj = dstObj.If(storage.Conditions{MetagenerationMatch: *req.DstMetaGenerationPrecondition})
+	}
+
+	// Converting the req.Sources list to a list of storage.ObjectHandle as expected by the Go Storage Client.
+	var srcObjList []*storage.ObjectHandle
+	for _, src := range req.Sources {
+		currSrcObj := b.bucket.Object(src.Name)
+		// Switching to requested Generation of the object.
+		if src.Generation != 0 {
```

Comment:
can src.Generation be nil, if yes what happens if it is nil

Reply:
No it can't be nil, it has int64 var type.